gax/src/main/java/com/google/api/gax/retrying/BasicRetryingFuture.java

```diff
@@ -166,6 +170,18 @@ void handleAttempt(Throwable throwable, ResponseT response) {
             retryAlgorithm.createNextAttempt(throwable, response, attemptSettings);
         boolean shouldRetry = retryAlgorithm.shouldRetry(throwable, response, nextAttemptSettings);
         if (shouldRetry) {
+          // Log retry info
+          if (LOG.isLoggable(Level.FINEST)) {
+            LOG.log(
+                Level.FINEST,
+                "Retrying with:\n{0}\n{1}\n{2}\n{3}",
+                new Object[] {
+                  "retryCallableMethod: " + callable.getClass().getEnclosingMethod().getName(),
```

Comment:
`geEnclosingMethod()` may return `null` if callable is not a anonymous class, resulting to NPE on `.getName()` call.

Reply:
done